Improve spacing, line splitting, hyphenation

Nyx102 opened the pull request:
I have found some information after looking through Volume 1:

- There was not a single instance where an ellipsis was split (word…word -> word…\nword). There were two instances where the ellipsis was the last character in the line and it was very close to the edge, but that was because it was the end of the sentence.
- There are very few cases where em dashes are split (word—word -> word—\nword).
- The lines with the biggest spaces that I have found are on page 122 and 33.

So based on this, we shouldn't actually make the em dashes and ellipses splittable. So in this pull request, I have modified the TeX file with some changes that should hopefully improve the spacing. Some before/after pages that I looked at are page 6 (screamed, large space before Rippling), 16 (Beasts), 131 (anything).

With this, I have also removed the splittable em dashes and ellipses, since it appears to work fine without them.

There are some errors from `microtype` because of `minimal` as the document class. I'm not sure why the document class is minimal, but that's another topic.

This appears to work well after looking through the PDF, but I cannot be sure that this does not introduce any weird things, so it probably needs closer inspection.

Nyx102 committed:
Improve spacing

Nyx102 commented:
I'm not sure if this should be a draft. I'm probably not going to change much about this; it's just that I'm not sure if there are any problems.

WorldEnd-Admin commented:
A few thoughts:

1. The document class is minimal just because it seemed most appropriate for a project where we're adding in the formatting, margins, font, etc manually. When I started this project I wasn't as aware of all the ramifications of using minimal as I am now. I'm not sure how much effort it is to switch, hopefully very little.
2. Even if the splittable em dash and ellipsis happen to not come up very often, are you sure we should disable them? I think that we'd still rather split on an existing ellipsis or em dash compared to adding a hyphen to a word, if it ever comes up. Were they actually causing any issues?

Nyx102 commented:
The splittable em dashes and ellipsis weren't causing any problems as far as I remember. But there were other issues, like big spaces and normal words overflowing, and on page 6, the em dash goes over the margin even after splitting.

So I added stuff to remove that, but after doing so, I removed the splittable em dash and ellipsis and it seemed to work fine.

I can't look at the document now, but if I remember correctly, it didn't add a hyphen to the lines I was comparing that had the em dashes and ellipses.

Nyx102 commented:
Em dashes are still split sometimes, it's just way less now.

Nyx102 commented:
Okay, so I compared a few dozen pages of the before and after.

It removes some hyphens and adds some hyphens, but overall I don't think it increases the amount of hyphens. It decreases some large spaces by stretching each space a little instead of a really big one. That does cause the spaces to be a little larger than they were before sometimes.

WorldEnd-Admin commented:
Tomorrow I'll do some tests to see how the automatic breaking at ellipsises and em dashes works. I'm curious to see whether it splits on them, and whether it knows to only split after them and not before. If it does then I'm fine with removing the splittable dash stuff, but if not then I'd prefer to keep it just to give TeX more opportunities for splitting.

WorldEnd-Admin commented:
Sorry this took me so long to get to, I had a crazy week. The new line/word splitting formatting you added (I assume polyglossia doing it) is able to split em dashes, but not on ellipses. So I added back in the python code to mark ellipses as splittable
